src/lang.rs

```diff
@@ -121,6 +88,58 @@ impl<T> Positioned<T> {
     }
 }
 
+/// Methods for converting a Span to different ranges over the span
+trait ToRange {
+    /// Return the entire Span as a Range
+    fn to_range(&self) -> Range<usize>;
+
+    /// Return a range from the current offset in the Span to the "sync point".  For the
+    /// angle-grinder syntax, the sync points are the vertical bar, end-of-query, and closing
+    /// braces/parens/etc...
+    fn to_sync_point(&self) -> Range<usize>;
+
+    /// Return a range from the current offset in the Span to the next whitespace
+    fn to_whitespace(&self) -> Range<usize>;
+}
```

Comment:
do we actually need this trait? Seems like we could just add these methods directly to Span

Reply:
I think it's needed since you can't `impl` methods on a type that is defined outside of the crate.